File: spoj/par.c
#include "stdio.h"

int main(void) {

    int c = 1, n, i, a, b;
    char j1[11], j2[11];
    scanf("%d", &n);
    while (n != 0) {
    	printf("Teste %d\n", c);
    	
    	scanf("%s\n%s", &j1, &j2);
    		
    	for (i = 0; i < n; i++) {
    		scanf("%d %d", &a, &b);
    		if ((a + b) % 2 == 0) {
    			printf("%s\n", j1);
    		}
    		else {
    			printf("%s\n", j2);
    		}
    	}
    	
    	printf("\n");
    	scanf("%d", &n);	
    	c++;
    }
    
    return 0;
}

File: spoj/par.py
# http://br.spoj.com/problems/PAR/

c = 1
n = int(input())
while n is not 0:
	print('Teste {}'.format(c))
	p1, p2 = input(), input()
	for x in range(n):
		n1, n2 = map(int, input().split())
		print(p1 if (n1 + n2) % 2 == 0 else p2)
	print()
	n = int(input())
	c += 1
